inbox/api/filtering.py

```diff
@@ -128,13 +127,17 @@ def threads(namespace_id, subject, from_addr, to_addr, cc_addr, bcc_addr,
                                started_after, last_message_before,
                                last_message_after, subject)
 
-    query = _threads_join_category(query, namespace_id, in_)
     query = query.filter(*filters)
+
     for subquery in _threads_subqueries(namespace_id, from_addr, to_addr,
                                         cc_addr, bcc_addr, any_email, filename,
-                                        unread, starred, db_session):
+                                        unread, starred, db_session, in_):
         query = query.filter(Thread.id.in_(subquery))
 
+    query = query.order_by(desc(Thread.recentdate)).limit(limit)
+    if offset:
+        query = query.offset(offset)
+
```

Comment:
Hmm, do we want to apply these to count queries? Otherwise this block should probably move back down to where it was